app/src/main/java/org/android/go/sopt/MyAdapter.kt

```diff
@@ -3,24 +3,23 @@ package org.android.go.sopt
 import android.content.Context
 import android.view.LayoutInflater
 import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import org.android.go.sopt.databinding.ItemMusicBinding
 
-class MyAdapter(context: Context) : RecyclerView.Adapter<MyAdapter.MusicViewHolder>() {
+class MyAdapter(context: Context) : ListAdapter<Music, MyAdapter.MusicViewHolder>(diffUtil) {
     private val inflater by lazy { LayoutInflater.from(context) }
 
-    private var itemlist: List<Music> = emptyList()
-
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MusicViewHolder {
         val binding = ItemMusicBinding.inflate(inflater, parent, false)
         return MusicViewHolder(binding)
     }
 
-    override fun getItemCount() = itemlist.size
 
     override fun onBindViewHolder(holder: MusicViewHolder, position: Int) {
-        holder.onBind(itemlist[position])
+        holder.onBind(currentList[position])
```

Comment:
```suggestion
        holder.onBind(getItem[position])
```
ListAdapter 내부에서 리스트 아이템을 관리하면서 getItemCount() 메서드가 재정의 되어 있고, 리스트의 아이템을 가져오는 getItem() 메서드가 생겨서 이렇게 적을 수도 있습니다!